
  <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
  <html xmlns="http://www.w3.org/1999/xhtml">
  <head>
<!--  <script src="http://bioinfo.bti.cornell.edu/js/analytics.js" type="text/javascript"></script>
#  <script src="http://bioinfo.bti.cornell.edu/tool/itak/js/jquery.js" type="text/javascript"></script>
#  <script src="http://bioinfo.bti.cornell.edu/tool/itak/js/search.js" type="text/javascript"></script>-->

  <title>iTAK - Plant Transcription factor & Protein Kinase Identifier and Classifier</title>
  <link href="http://bioinfo.bti.cornell.edu/tool/itak/style.css" rel="stylesheet" type="text/css" media="screen" />
  <link rel="stylesheet" href="http://bioinfo.bti.cornell.edu/tool/itak/jquery.treeview.css" />
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.min.js"></script>
  <script src="http://bioinfo.bti.cornell.edu/tool/itak/js/jquery.cookie.js"></script>
  <script src="http://bioinfo.bti.cornell.edu/tool/itak/js/jquery.treeview.js"></script>
  <script src="http://bioinfo.bti.cornell.edu/tool/itak/js/d3.min.js"></script>
  <!--<script src="http://bioinfo.bti.cornell.edu/tool/itak/js/search.js" type="text/javascript"></script>-->
  <script type="text/javascript" src="http://bioinfo.bti.cornell.edu/tool/itak/js/demo.js"></script>

  <style type="text/css">
  .menu {
    font-family: verdana, sans-serif; 
    /*position:relative; */
    font-size: 13px;
    padding-bottom:10px;
    padding-left:5px;
    padding-right: 5px;
  }
  .menu ul {
    padding:0; 
    margin:0;
    list-style-type: none;
  }
  .menu ul li {
    float:left;
    position:relative;
  }
  .menu ul li a, .menu ul li a:visited {
    display:block; 
    text-decoration:none;
    width:139px;
    height:30px;
    color:#fff;
    border:1px solid #fff; 
    border-width:1px 1px 0 0; 
    background:#3b3b3b;
    text-align:center;
    line-height:30px;
  }

  /* specific to non IE browsers */
  .menu ul li:hover a {
    color:#fff; 
    background:#bd8d5e;
  }

  </style>
  <!--[if lte IE 6]>
  <style type="text/css">
  .menu ul li a:hover {
    color:#fff; 
    background:#bd8d5e;
  } 
  <![endif]-->

  </head>
  <body>
  
  <div id="wrapper">
    <div id="logo">
      <p><b>iTAK</b> - Plant Transcription factor & Protein Kinase Identifier and Classifier</p>
    </div>
    <hr />
    <!-- end #logo -->
    <div id="header">
    <div class="menu">
      <ul id=navigation>
      <li><a href=/cgi-bin/itak/index.cgi><b>Home</b></a></li>
      <li><a href=/cgi-bin/itak/itak.cgi><b>iTAK server</a></b></li>
      <li><a href=/cgi-bin/itak/db_home.cgi><b>Database</b></a></li>
      <li><a href=/cgi-bin/itak/download.cgi><b>Download</b></a></li>
      <li><a href=/cgi-bin/itak/help.cgi><b>Help</b></a></li>
      <li><a href=/cgi-bin/itak/contact.cgi><b>Contact</b></a></li>
      </ul>
    </div>
    <!-- end #menu -->

  </div>
  <!-- end #header -->
  <!-- end #header-wrapper -->

  <div id="page">
  <center>

<div class=rec_left>
<h3 align=center>iTAK (v1.5)</h3><br>

<table width=800 align=center>
 <tr><td>
  <b><ul>
  <li><a href=#requirement>System requirement and dependencies</a></li>
  <li><a href=#release>Release notes</a></li>
  <li><a href=#install>Installation</a></li>
  <li><a href=#run>Running iTAK</a></li>
  <li><a href=#output>iTAK output</a></li>
  <li><a href=#performance>Performance</a></li>
  </ul></b>
 </td></tr>
</table>
<br><br>

<a name=requirement></a>
<h3>System requirement and dependencies</h3><br>
<h4>>>iTAK server</h4><br>
<p>iTAK server has been tested under firefox, IE, and Safari. If it's incompatible 
with your browser, please feel free to <a href=contsci.cgi>contact us</a>.</p>

<h4>>>iTAK standalone program</h4><br>
<ul>
  <li>64-bit Linux system</li>
  <li>Perl version 5.10.0 or higher (required). <a href=http://www.perl.org 
  target=_blank>Perl</a> was installed by default on most Linux systems</li>
  <li>BioPerl version 1.006 or higher (required). Please check <a 
  href=http://www.bioperl.org target=_blank>http://www.bioperl.org</a> and <a 
  href=http://www.bioperl.org/wiki/Installing_BioPerl target=_blank>
  wiki/Installing_BioPerl</a> for more details on installation of BioPerl.</li>
  <li><a href=http://hmmer.janelia.org>HMMER3</a> (required). Provided in iTAK.</li>
</ul>

<br>
<a name=release></a>
<h3>Release notes</h3><br>
<ul>
  <li>iTAK v1.5 - Feb-x-2015<br>
   The version 1.5 is a important update, and it has many advantages compare to previous versions. <br> 
&nbsp;&nbsp;&nbsp;--&nbsp;update a comperehensive set of rules for identification of transcription factor <br>
&nbsp;&nbsp;&nbsp;--&nbsp;shrink the database and significantly accelerate the speed <br>
&nbsp;&nbsp;&nbsp;--&nbsp;automaticially detect the alphabet of input sequences <br>
&nbsp;&nbsp;&nbsp;--&nbsp;provie options of frames for nucleotide translation <br>
&nbsp;&nbsp;&nbsp;--&nbsp;only 64-bit version will be provide from iTAK v1.5. If you want use iTAK on 32-bit machine, please check <a href=#install>installation</a> for details. <br><br>
  </li>

  <li>iTAK v1.4 - Jan-19-2014<br>
   update the hmmscan to version 3.1, wich is 2x faster than hmm 3.0. <br> 
   add new classification system for plant protein kinase, which was built by <a href="ttp://www.ncbi.nlm.nih.gov/pubmed/22889912">Lehti-Shiu and Shiu, 2012</a><br><br>
  </li>

  <li>iTAK v1.3 - Jun-22-2013 <br>
   update Pfam database to version 27, and fix some small bugs <br><br>
  </li>

  <li>iTAK v1.2 - Aug-26-2011<br>
   report unusual sequences<br><br>
  </li>

  <li>iTAK v1.1 - Jun-03-2011<br>
   remove unsignificiant domain using GA score<br><br>
  </li>

  <li>iTAK v1.0 - Dec-14-2010</li>
</ul>
<br>

<a name=install></a>
<h3>Installation</h3><br>
<h4>>>iTAK standalone program</h4><br>
<p>Installation of iTAK is more straightforward than previous. Everything you need is to download 
the lastest version of iTAK and uncompress it. <br>

<table border=1 style="border-collapse: collapse" cellpadding=5 bordercolor=#555555 width=480>
<tr><td bgcolor=#EEEEEE><font face=courier><b>
        shell$ tar -xzvf iTAK-1.5.tar.gz
</b></font></td>
</tr></table>

<br>
It will generate a folder named "iTAK-1.x" and we call this folder "iTAK home folder". iTAK home 
folder includes three subfolders, a "bin" folder containing the HMMER3 executable, a "database" 
folder containing the domain database files and a "doc" folder containing the program documentation 
files. The home folder also contains a perl script, iTAK.pl, which is the core script to run 
the whole iTAK pipeline. 
<br>

<br>
After that, please run test sequences to initialize databases.
<br><br>

<table border=1 style="border-collapse: collapse" cellpadding=5 bordercolor=#555555 width=480>
<tr><td bgcolor=#EEEEEE><font face=courier><b>
        shell$ cd iTAK-1.5 <br>
        shell$ perl iTAK.pl test_seq
<br></b></font></td>
</tr></table>

<br>
<br>
<b>For 32-bit user</b><br>
Only 64-bit version will be provide from iTAK v1.5. The only difference between 64-bit and 32-bit 
iTAK is recruitment of different versions of HMMER. To run iTAK on 32-bit machine, please replace 
the HMMER with ia32 version.
</p>
<br>

<a name=run></a>
<h3>Running iTAK</h3><br>
<h4>>>standalone version</h4><br>
<b><font color=#A208A0>Quick Start</font></b><br><br>
<ol>
<li>Put the protein or nucleotide sequence file in FASTA format in iTAK home folder </li>
<li>Go to iTAK home folder and run iTAK with the following command (assuming the input file 
name is input_seq)</li>
<br>
<table border=1 style="border-collapse: collapse" cellpadding=5 bordercolor=#555555 width=610>
<tr><td bgcolor=#EEEEEE><font face=courier><b>shell$ perl iTAK.pl input_seq</b></font></td>
</tr></table>
<br>
<li>The program will generate an output folder named input_seq_output which contains all the 
output files. See below for the description of the output files.</li>
</ol><br>

<b><font color=#A208A0>Parameters</font></b><br><br>

<table border=0 cellspacing=0 cellpadding=5>
<tr valign=top><th width=15>-f</th><th>[String]</th><td >Frame usage of nucleotie translation, could be: <br>
	6:  6 frame translation (default) <br>
	3F: 3 frame translation with forward strand <br>
	3R: 3 frame translation with reverse strand
	</td></tr>
<tr valign=top><th>-a</th><th>[Integer]</th><td >number of CPUs used for hmmscan (default = 1)</td></tr>
<tr valign=top><th>-o</th><th>[String]</th><td >Name of the output directory (default = "input file name" + "_output")</td></tr>
</table>

<br>
<br>

<b>obsoleting parameters</b> <br>
<table border=0 cellspacing=0 cellpadding=5>
<tr valign=top><th width=15>-i</th><th>[String]</th><td >Name of the input sequence file in
FASTA format  (<i>required</i>)</td></tr>
<tr valign=top><th>-s</th><th>[String]</th><td >Type of input sequence. 'p' for protein
sequences | 'n' for nucleotide sequences. (default = p)
</td></tr>
<tr valign=top><th>-m</th><th>[String]</th><td>Type of analysis ('t' for TF identification
| 'p' for PK identification | 'b' for both) (default = b) </td></tr>
</table>
<br><br>

<a name=output></a>
<h3>iTAK output</h3><br>
<h4>>>iTAK server</h4><br>
<p>The output of iTAK web server contains tables which list the identified transcription factors, transcription regulators, and 
protein kinases, as well as their corresponding families (see figure below). The detailed alignments of each protein sequence 
against protein domains are also provided.<br><br>
<img src=http://bioinfo.bti.cornell.edu/tool/itak/images/outb.png border=1><br><br>
An example output is provided <a href=/cgi-bin/itak/itak.cgi?rid=PTEST target=_blank>here</a>.

<h4>>>iTAK standalone program</h4><br>
<p>Eight files will be generated in the output directory - "input_seq_output" (assuming the input file name is "input_seq").<br>

<ol>
	<li><b>tf_sequence.fasta:</b> sequences of all identified transcription factors (FASTA format).</li>
	<li><b>tf_classification.txt:</b> classification of all identified transcription factors. A tab-delimited txt file 
containing sequence IDs and their corresponding transcription factor families.</li>
	<li><b>tf_alignment.txt:</b> A tab-delimited txt file containing alignments of all identified transcription factors 
to the protein domain database (from hmmscan analysis).</li>
	<li><b>pk_sequence.fasta:</b> sequences of all identified protein kinases (FASTA format).</li>
	<li><b>PPC_classification.txt:</b> classification of all identified protein kinases. A tab-delimited txt file containing sequence 
IDs and their corresponding protein kinase families.</li>
	<li><b>PPC_alignment.txt:</b> A tab-delimited txt file containing alignments of all identified protein kinases to 
the protein domain database (from hmmscan analysis).</li>
	<li><b>Shiu_classification.txt:</b> classification of all identified protein kinases. A tab-delimited txt file containing sequence     
IDs and their corresponding protein kinase families.</li>
	<li><b>Shiu_alignment.txt:</b> A tab-delimited txt file containing alignments of all identified protein kinases to
the protein domain database (from hmmscan analysis).</li>
</ol>
</p>
<br>
<a name=performance></a>
<h3>Performance</h3><br>
We run iTAK on a single CPU on a 64-bit laptop with i5-3320M CPU @ 2.60GHz. It took aout 27 minutes to 
identify both transcription factors and protein kinases from 33,410 Arabidopsis protein sequences (<a 
href=ftp://ftp.arabidopsis.org/home/tair/Sequences/blast_datasets/TAIR10_blastsets/ target=_blank>TAIR10 release</a>). <br>

</div>
 
  <div style="clear: both;">&nbsp;</div>
  <br><br>
  </center>
  </div>
  <!-- end #page -->
  <div id="footer">
    <p><a href="http://bioinfo.bti.cornell.edu">Fei Bioinformatics Lab</a>, Boyce Thompson Institute and USDA Robert W. Holley Center</p>
  </div>
  <!-- end #footer -->
  </div>
  </body>
  </html>
  
